static/js/iframe-prod.js

```diff
@@ -1,5 +1,15 @@
-import { generateIFrame } from './iframe-common';
+import { generateIFrame, sendToIframe } from './iframe-common';
 import InjectedData from './models/InjectedData';
+import generateInitAnswersFrame from './utils/generateInitAnswersFrame';
+import { RuntimeConfig } from './runtime-config';
 
-const prodDomain = new InjectedData().getProdDomain()
-generateIFrame(prodDomain);
+const runtimeConfig = new RuntimeConfig();
+window.setRuntimeConfig = (key, value) => {
+  runtimeConfig.set(key, value);
+  sendToIframe({ runtimeConfig: runtimeConfig.getObject() });
+}
+
+const prodDomain = new InjectedData().getProdDomain();
+generateIFrame(prodDomain, runtimeConfig);
+
+window.initAnswersExperienceFrame = generateInitAnswersFrame(runtimeConfig);
```

Comment:
Here and in the other iFrame script, are we waiting for them to set the `runtimeConfig` before calling `initAnswers`? If so, it's a little hard to follow how that's happening. For my own edification, could you explain that flow?

Reply:
Ah, after refactoring the 'generateInitAnswersFrame' naming is no longer entirely accurate. I will update it. What happens is that the iframe is always generated immediately. If there were any data attributes set, those will be included in the runtime config during initialization. Next, the `initAnswersExperienceFrame` function is put on the window which updates the runtime config and sends the message to the iframe that the Answers experience should initialize